hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -454,17 +440,32 @@ private void writeTableMetadataForTableServices(HoodieTable<T, JavaRDD<HoodieRec
       }
       metaClient.reloadActiveTimeline();
 
-      // re-bootstrap metadata table if required
       initializeMetadataTable(Option.of(instantTime));
```

Comment:
for multi-writer scenarios, I think it's important that this is within the log

Reply:
right, taken care in https://github.com/apache/hudi/pull/4124